stock_cancel/models/stock.py

```diff
@@ -47,15 +48,27 @@ def action_revert_done(self):
             if picking.invoice_id:
                 raise exceptions.Warning(
                     _('Picking %s has invoices!') % (picking.name))
+            for move in picking.move_lines:
```

Comment:
You can save one loop with: `for quant in picking.mapped('move_lines.quant_ids'):`